nemo/collections/nlp/data/datasets/sgd_dataset/prediction_utils.py

```diff
@@ -93,125 +201,241 @@ def get_predicted_dialog_ret_sys_act(dialog, all_predictions, schemas, eval_debu
 
                 # Add prediction for user goal (slot values).
                 # Categorical slots.
-                categorical_slots_dict = {}
-                non_categorical_slots_dict = {}
-
-                predictions["cat_slot_status_p"] = predictions["cat_slot_status_p"].cpu().numpy()
-                predictions["cat_slot_status"] = predictions["cat_slot_status"].cpu().numpy()
-                predictions["cat_slot_value"] = predictions["cat_slot_value"].cpu().numpy()
-                predictions["cat_slot_value_p"] = predictions["cat_slot_value_p"].cpu().numpy()
-
-                predictions["noncat_slot_status_p"] = predictions["noncat_slot_status_p"].cpu().numpy()
-                predictions["noncat_slot_status"] = predictions["noncat_slot_status"].cpu().numpy()
-                predictions["noncat_slot_p"] = predictions["noncat_slot_p"].cpu().numpy()
-
-                predictions["noncat_alignment_start"] = predictions["noncat_alignment_start"].cpu().numpy()
-                predictions["noncat_alignment_end"] = predictions["noncat_alignment_end"].cpu().numpy()
-                predictions["cat_slot_status_GT"] = predictions["cat_slot_status_GT"].cpu().numpy()
-                predictions["noncat_slot_status_GT"] = predictions["noncat_slot_status_GT"].cpu().numpy()
-
+                categorical_slots_dict = OrderedDict()
+                non_categorical_slots_dict = OrderedDict()
                 for slot_idx, slot in enumerate(service_schema.categorical_slots):
-                    # debugging info
                     cat_slot_status_num += 1
-                    categorical_slots_dict[slot] = (
-                        predictions["cat_slot_status_GT"][slot_idx],
-                        predictions["cat_slot_status"][slot_idx],
-                        predictions["cat_slot_status_p"][slot_idx],
-                        service_schema.get_categorical_slot_values(slot)[predictions["cat_slot_value"][slot_idx]],
-                        predictions["cat_slot_value_p"][slot_idx],
-                    )
-
-                    if predictions["cat_slot_status_GT"][slot_idx] == predictions["cat_slot_status"][slot_idx]:
-                        cat_slot_status_acc += 1
 
                     slot_status = predictions["cat_slot_status"][slot_idx]
+                    ext_value = None
```

Comment:
what does `ext_value` stand for?

Reply:
It stands for extracted value. Better name?

Reply:
extracted_value ?

Reply:
Changed.